inc/namespace.php

```diff
@@ -0,0 +1,314 @@
+ */
+function get_hash_for_script( string $handle ) : ?string {
+	global $wp_scripts;
+	return get_hash_for_asset( $wp_scripts, $handle );
+}
+
+/**
+ * Get the integrity hash for a style.
+ *
+ * Use `set_hash_for_style()` to set the integrity hash for a style.
+ *
+ * @param string $handle Unique style handle.
+ * @return string|null Integrity hash if set, null otherwise.
+ */
+function get_hash_for_style( string $handle ) : ?string {
+	global $wp_styles;
+	return get_hash_for_asset( $wp_styles, $handle );
+}
+
+/**
+ * Get the integrity hash for an asset.
+ *
+ * Use `set_hash_for_asset()` to set the integrity hash for an asset.
+ *
+ * @param WP_Dependencies $dependencies Dependency registry to use.
+ * @param string $handle Unique asset handle.
+ * @return string|null Integrity hash if set, null otherwise.
+ */
+function get_hash_for_asset( WP_Dependencies $dependencies, string $handle ) : ?string {
+	return $dependencies->get_data( $handle, INTEGRITY_DATA_KEY ) ?? null;
+}
+
+/**
+ * Set the integrity hash for a script.
+ *
+ * @param string $handle Unique script handle.
+ * @param string $hash Integrity hash (in format "<algo>-<hash>").
+ * @return boolean True if the hash was set correctly, false otherwise.
+ */
+function set_hash_for_script( string $handle, string $hash ) : bool {
+	global $wp_scripts;
+	return set_hash_for_asset( $wp_scripts, $handle, $hash );
+}
+
+/**
+ * Set the integrity hash for a stylesheet.
+ *
+ * @param string $handle Unique style handle.
+ * @param string $hash Integrity hash (in format "<algo>-<hash>").
+ * @return boolean True if the hash was set correctly, false otherwise.
+ */
+function set_hash_for_style( string $handle, string $hash ) : bool {
+	global $wp_styles;
+	return set_hash_for_asset( $wp_styles, $handle, $hash );
+}
+
+/**
+ * Set the integrity hash for an asset.
+ *
+ * @param WP_Dependencies $dependencies Dependency registry to use.
+ * @param string $handle Unique asset handle.
+ * @param string $hash Integrity hash (in format "<algo>-<hash>").
+ * @return boolean True if the hash was set correctly, false otherwise.
+ */
+function set_hash_for_asset( WP_Dependencies $dependencies, string $handle, string $hash ) : bool {
+	return $dependencies->add_data( $handle, INTEGRITY_DATA_KEY, $hash );
+}
+
+/**
+ * Output the integrity hash for a script.
+ *
+ * This is automatically added to the `script_loader_tag` filter. Use
+ * `set_hash_for_script()` to set the integrity hash for a script.
+ *
+ * @param string $tag Script HTML tag.
+ * @param string $handle Unique script handle.
+ * @return string Script tag with `integrity` attribute set if available.
+ */
+function output_integrity_for_script( string $tag, string $handle ) : string {
+	$hash = get_hash_for_script( $handle );
+	if ( empty( $hash ) ) {
+		return $tag;
+	}
+
+	// Insert the attribute.
+	$tag = str_replace(
+		"type='text/javascript' src='",
+		sprintf(
+			"type='text/javascript' integrity='%s' src='",
```

Comment:
Do you think it's worth explicitly setting the `crossorigin` attribute to? Defaults to "anonymous" which is what we want generally.

Reply:
I think that's too much of an assumption to make with integrity hashes only, but we could add it separately.